ci/container/container-run.sh

```diff
@@ -160,21 +162,32 @@ mkdir -p ~/.{aws,ssh,cache,local/share/fish} && touch ~/.{zsh,bash}_history
 PODMAN_RUN_USR_ARGS=()
 if [ -f "$HOME/.container-run.conf" ]; then
     # conf file with user's custom PODMAN_RUN_USR_ARGS
-    echo "Sourcing user's ~/.container-run.conf"
+    echo "Sourcing user's ~/.container-run.conf" >&2
     source "$HOME/.container-run.conf"
 fi
 
-# privileged rootful podman is required due to requirements of IC-OS guest build
-# additionally, we need to use hosts's cgroups and network
+# Omit -t if not a tty.
+# Also shut up logging, because podman will by default log
+# every byte of standard output to the journal, and that
+# destroys the journal + wastes enormous amounts of CPU.
+# I witnessed journald and syslog peg 2 cores of my devenv
+# when running a simple cat /path/to/file.
+if tty >/dev/null 2>&1; then
+    tty=" -t --log-driver=none"
+else
+    tty="--log-driver=none"
+fi
+# Privileged rootful podman is required due to requirements of IC-OS guest build;
+# additionally, we need to use hosts's cgroups and network.
 if [ $# -eq 0 ]; then
     set -x
-    sudo podman "${PODMAN_ARGS[@]}" run --pids-limit=-1 -it --rm --privileged --network=host --cgroupns=host \
+    sudo podman "${PODMAN_ARGS[@]}" run --pids-limit=-1 -i $tty --rm --privileged --network=host --cgroupns=host \
         "${PODMAN_RUN_ARGS[@]}" ${PODMAN_RUN_USR_ARGS[@]} -w "$WORKDIR" \
         "$IMAGE" ${USHELL:-/usr/bin/bash}
     set +x
 else
     set -x
-    sudo podman "${PODMAN_ARGS[@]}" run --pids-limit=-1 -it --rm --privileged --network=host --cgroupns=host \
+    sudo podman "${PODMAN_ARGS[@]}" run --pids-limit=-1 -i $tty --rm --privileged --network=host --cgroupns=host \
         "${PODMAN_RUN_ARGS[@]}" "${PODMAN_RUN_USR_ARGS[@]}" -w "$WORKDIR" \
         "$IMAGE" "$@"
     set +x
```

Comment:
to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) you could consider:
```
if [ $# -eq 0 ]; then
    COMMAND="${USHELL:-/usr/bin/bash}"
else
    COMMAND="$@"
fi

set -x
sudo podman "${PODMAN_ARGS[@]}" run --pids-limit=-1 -i $tty --rm --privileged --network=host --cgroupns=host \
        "${PODMAN_RUN_ARGS[@]}" "${PODMAN_RUN_USR_ARGS[@]}" -w "$WORKDIR" \
        "$IMAGE" "$COMMAND"
set +x
```

Reply:
That mangles whitespace of arguments.  Look:

```
 manuel   container-run-fixes ✚ 1 … 1 ⚑ 6  ~  src  ic  cat tester
if [ $# -eq 0 ]; then
    COMMAND="${USHELL:-/usr/bin/bash}"
else
    COMMAND="$@"
fi

set -x
echo "$COMMAND"
set +x
 manuel   container-run-fixes ✚ 1 … 1 ⚑ 6  ~  src  ic  bash -x tester echo "this has spaces"
+ '[' 2 -eq 0 ']'
+ COMMAND='echo this has spaces'
+ set -x
+ echo 'echo this has spaces'
echo this has spaces
+ set +x
```

Reply:
Good catch. I think this will be fixed by using arrays:
```
if [ $# -eq 0 ]; then
    COMMAND=("${USHELL:-/usr/bin/bash}")
else
    COMMAND=("$@")
fi

set -x
sudo podman "${PODMAN_ARGS[@]}" run --pids-limit=-1 -i $tty --rm --privileged --network=host --cgroupns=host \
        "${PODMAN_RUN_ARGS[@]}" "${PODMAN_RUN_USR_ARGS[@]}" -w "$WORKDIR" \
        "$IMAGE" "${COMMAND[@]}"
set +x
```

Reply:
```
if [ $# -eq 0 ]; then
    COMMAND="${USHELL:-/usr/bin/bash}"
else
    COMMAND="$@"
fi
```

This seems like a clever idea initially, but it doesn't preserve the quotes of each argument in `$@` when moving it into `$COMMAND`.  So `container-run echo "abc def"` gets translated into `container-run echo abc def`.

Sorry, NAK.

Reply:
I tried the suggestion, it didn't work.  What can I say.